gleam.toml

```diff
@@ -14,7 +14,7 @@ links = [
 gleam_stdlib = "~> 0.31"
 gleam_http = "~> 3.0"
 gleam_otp = "~> 0.7"
-cowboy = "~> 2.0"
+cowboy = "~> 2.10.0"
```

Comment:
You've downgraded the requirement here to be very pessimistic. Before it was installing `2.x`, now it's `2.10.x`. Could you revert this please? 🙏

Reply:
Yeah I addressed this in the third dot point, but basically with cowboy 2.11.0 they changed the way they handled cookie headers. In particular, you get this error when running the tests:
```
=ERROR REPORT==== 17-Mar-2024::16:54:14.917416 ===
Ranch listener {gleam_cowboy,#Ref<0.2594036081.850657288.229963>}, connection process <0.573.0>, stream 1 had its request process <0.574.0> exit with reason {response_error,invalid_header,'Response cookies must be set using cowboy_req:set_resp_cookie/3,4.'} and stacktrace [{cowboy_req,reply,4,[{file,"/Users/bradlewis/repos/cowboy/build/dev/erlang/cowboy/src/cowboy_req.erl"},{line,813}]},{gleam_cowboy_native,init,2,[{file,"/Users/bradlewis/repos/cowboy/build/dev/erlang/gleam_cowboy/_gleam_artefacts/gleam_cowboy_native.erl"},{line,22}]},{cowboy_handler,execute,2,[{file,"/Users/bradlewis/repos/cowboy/build/dev/erlang/cowboy/src/cowboy_handler.erl"},{line,37}]},{cowboy_stream_h,execute,3,[{file,"/Users/bradlewis/repos/cowboy/build/dev/erlang/cowboy/src/cowboy_stream_h.erl"},{line,306}]},{cowboy_stream_h,request_process,3,[{file,"/Users/bradlewis/repos/cowboy/build/dev/erlang/cowboy/src/cowboy_stream_h.erl"},{line,295}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,241}]}]
```
I'll have a quick look into seeing if I can update the lib to support `cowboy_req:set_resp_cookie` instead to keep the requirement as is.

Reply:
Roger. Let's go for `~> 2.11` to be on the right side of that breaking change.

Reply:
After a bit of trial and error I managed to get it to work. I needed to update the `CowboyRequest` object directly, which is the path that was suggested here: https://github.com/ninenines/cowboy/pull/1624#issuecomment-1915324578 and the solution chosen by the elixir cowboy project. 

This lets us keep the requirement back to just `~> 2.0`. Let me know what you think of this.

FWIW I did look into using the `cowboy_req:set_resp_cookies` function, but I was unable to get that to work, both due to parsing the cookie string dropping some information, and not being able to pass the opts easily.